integrations/helm/chartsync/chartsync.go

```diff
@@ -403,6 +406,11 @@ func (chs *ChartChangeSync) DeleteRelease(fhr fluxv1beta1.HelmRelease) {
 
 // getNamespaces gets current kubernetes cluster namespaces
 func (chs *ChartChangeSync) getNamespaces() ([]string, error) {
+	if chs.namespace != "" {
```

Comment:
The other branch here asks the API server, whereas this just trusts what it's been told. This might break assumptions made by callers of this code, e.g., that the namespace(s) returned definitely exist.

Reply:
This gets called by `getCustomResources` and if the namespace is missing the error is caught when fetching a HR. The error will tell you that there is no HR in that namespace. Also if the namespace is missing, Flux will not apply the HR.

Reply:
That may be how it works now. Can you guarantee it will always be that way?
Actually I have a feeling that there's only one call site for this (`getCustomResources`); if that's the case, you may as well inline it, and my objection won't matter.

Reply:
I've deleted `getNamespaces` and moved the code to `getCustomResources`